data_processors/pipeline/domain/workflow.py

```diff
@@ -230,12 +234,98 @@ def __init__(self, this_workflow):
     def must_associate_sequence_run(self):
         this_sqr = self.workflow.sequence_run
         if this_sqr is None:
-            raise ValueError(f"Workflow {self.workflow.type_name} wfr_id: '{self.workflow.wfr_id}' must be associated "
-                             f"with a SequenceRun. SequenceRun is: {this_sqr}")
+            raise WorkflowRuleError(f"Workflow {self.workflow.type_name} wfr_id: '{self.workflow.wfr_id}' must be "
+                                    f"associated with a SequenceRun. SequenceRun is: {this_sqr}")
         return self
 
     def must_have_output(self):
         # use case e.g. bcl convert workflow run must have output in order to continue next step(s)
         if self.workflow.output is None:
-            raise ValueError(f"Workflow '{self.workflow.wfr_id}' output is None")
+            raise WorkflowRuleError(f"Workflow '{self.workflow.wfr_id}' output is None")
+        return self
+
+
+class MetadataRuleError(ValueError):
+    pass
+
+
+class MetadataRule:
```

Comment:
`LabMetadataRule` for consistency with data model?
(less of an issue to LibraryRule, see other point)

Reply:
Ah haha. I thought about the other way round; wanted to rename `LabMetadata` to just `Metadata`.  But, yah, (sneak attempt fail; hehe) I will rename it to `LabMetadataRule` 😉